docs/blueprints/storage-node-downtime-tracking-with-audits.md

```diff
@@ -0,0 +1,153 @@
+}
+```
+
+### 5) Write to audit windows and evaluate node standing. 
+The overlay DB method `BatchUpdateStats` is where a node's audit reputation is updated by audit results. It iterates through each node, retrieving its dossier from the `nodes` table and updates its reputation. We also need to update and read information from the node dossier to set and evaluate a node's standing regarding audit windows, so this method is a good place to do this. 
+
+Add a new argument to `BatchUpdateStats` to pass in `AuditWindowConfig`. This gives us the window size parameter required to write to the audit windows table and the tracking period, grace period, and offline threshold parameters required to evaluate whether a node needs to be suspended, reinstated, or disqualified.
+
+For each node, take the `offline_suspended`, `offline_scored_at`, and `under_review` values from the node dossier. 
+Determine whether we need to evaluate the node's standing (see implementation step 2 above).
+If so, evaluate the node's current standing by scoring each window by finding what percentage of audits were offline per window, then finding the average score across all complete windows within the tracking period.
+
+```sql
+SELECT SUM(offline_score)/COUNT(*)
+FROM (
+    SELECT (offline/total) AS offline_score
+    FROM audit_windows
+    WHERE node_id = $1 AND window_start >= $2 AND window_start < $3
+) t
+```
+
+With this information, there are a number of conditions we need to evaluate:
+
+1) The current offline score is above the OfflineThreshold
+    
+    1a. The node is under review
+
+    - Reinstate the node if it is suspended.
+
+    - Check if the review period has expired. We can do this by taking the start boundary of the current window and subtracting the tracking period and grace period lengths from it. If this value is greater than the `under_review` value, this means that the node's review period has elapsed and the `under_review` column can be cleared.
+
+    1b. The node is not under review
+
+    - The node is in good standing and does not need to be updated.
+
+2) The current offline score is below the OfflineThreshold
+
+    2a. The node is under review
+
+    - Check if the review period has expired. If so, the `disqualified` column should be set (either to the current time, or the start boundary of the current window since that was the point at which all of the data had been collected)
+
+    - Suspend the node if it is not already suspended. 
+        
+    2b. The node is not under review
+
+    - Suspend the node and set its `under_review` column to the current time.
+
+
+After the evaluation is complete, insert or update the appropriate window in the audit_windows table.
+
+### 6) Implement email and node dashboard notifications of offline suspension
+
+### 7) Create a chore to delete audit windows after a certain time 
```

Comment:
we wouldn't need this if we did the protobuf approach. they'd automatically get cleaned up